pvlib/iam.py

```diff
@@ -748,3 +748,108 @@ def marion_integrate(function, surface_tilt, region, num=None):
         Fd = pd.Series(Fd, surface_tilt.index)
 
     return Fd
+
+
+def fedis(aoi, surface_tilt, n=1.5, n_ref=1.4585):
+    """
+    Determine the incidence angle modifiers (iam) for direct, diffuse sky,
+    and ground-reflected radiation using the FEDIS transmittance model.
+
+    The "Fresnel Equations" for Diffuse radiation on Inclined photovoltaic
+    Surfaces (FEDIS) [1]_ is an analytical solution of diffuse transmission
+    based on the rigorous integration of an alternate form of the
+    Fresnel equations. The approach leads to a simple yet accurate
+    relative transmittance model that reconciles the solar energy
+    sensed by pyranometers and PV panels.
+
+    Parameters
+    ----------
+    aoi : numeric
+        Angle of incidence. [degrees]
+
+    surface_tilt : numeric
+        Surface tilt angle measured from horizontal (e.g. surface facing
+        up = 0, surface facing horizon = 90). [degrees]
+
+    n : float, default 1.5
+        Refractive index of the PV cover.  The default value of 1.5
+        was used for an IMT reference cell in [1]_. [unitless]
+
+    n_ref : float, default 1.4585
+        Refractive index of the pyranometer cover. The default value
+        was used for a fused silica dome over a CMP22 in [1]_.
+
+    Returns
+    -------
+    iam : dict
+        IAM values for each type of irradiance:
+
+            * 'direct': radiation from the solar disc
+            * 'sky': radiation from the sky dome (zenith <= 90)
+            * 'ground': radiation reflected from the ground (zenith >= 90)
+
+    Notes
+    -----
+    This implementation corrects a typo in the reference regarding the sign
+    of the last polynomial term in Equation 5.
+
+    References
+    ----------
+    .. [1] Xie, Y., M. Sengupta, A. Habte, A. Andreas, "The 'Fresnel Equations'
+       for Diffuse radiation on Inclined photovoltaic Surfaces (FEDIS)",
+       Renewable and Sustainable Energy Reviews, vol. 161, 112362. June 2022.
+       :doi:`10.1016/j.rser.2022.112362`
+    """
+
+    # avoid undefined results for horizontal or upside-down surfaces
+    zeroang = 1e-06
+    surface_tilt = np.where(surface_tilt == 0, zeroang, surface_tilt)
```

Comment:
The need for this is a caution flag, to me. It addresses the divide by zero possiblity in the "/ (1 - cos(surface_tilt))" portion of the equation for the ground-reflected IAM equation.  I am not familiar with the reference. Is it possible that the integrand becomes singular as surface_tilt -> 0? If so, maybe there's an alternate way to show that the ground-reflected IAM converges to 0.

Reply:
I settled on this approach (lifted from `martin_ruiz_diffuse`) because it also prevents the obnoxious `RuntimeWarning: divide by zero encountered in true_divide` warnings from numpy.  I'm certainly happy to change to a better alternative if there is one.

Reply:
Here's an alternative: for `surface_tilt` < minimum, set `cuk = 20 w / 21` (limiting case of Eq. A8a in [1]), and set `cug = 0` (there should be no ground-reflected irradiance on a horizontal, upward-facing surface.